waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/DatabaseMappingUtils.java

```diff
@@ -0,0 +1,128 @@
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.hadoop.hive.metastore.api.TableMeta;
+import org.slf4j.Logger;
+
+import com.hotels.bdp.waggledance.mapping.model.DatabaseMapping;
+import com.hotels.bdp.waggledance.util.Whitelist;
+
+public class DatabaseMappingUtils {
+
+  public static final String MANUAL_RESOLUTION_TYPE = "MANUAL";
+  public static final String PREFIXED_RESOLUTION_TYPE = "PREFIXED";
+  private static final String INTERRUPTED_MESSAGE = "Interrupted ";
+  private static final String SLOW_METASTORE_MESSAGE = "Metastore was slow to respond so results are omitted";
+  private static final String NULL_POINTER_MESSAGE = "Got nullpointer";
+
+  public static boolean isWhitelisted(String databasePrefix, String database, Map<String, Whitelist> mappedDbByPrefix) {
+    Whitelist whitelist = mappedDbByPrefix.get(databasePrefix);
+    if ((whitelist == null) || whitelist.isEmpty()) {
+      // Accept everything
+      return true;
+    }
+    return whitelist.contains(database);
+  }
+
+  public static List<String> getMappedWhitelistedDatabases(List<String> databases, DatabaseMapping mapping,
+                                                           Map<String, Whitelist> mappedDbByPrefix) {
+    List<String> mappedDatabases = new ArrayList<>();
+    for (String database : databases) {
+      if (DatabaseMappingUtils.isWhitelisted(mapping.getDatabasePrefix(), database, mappedDbByPrefix)) {
+        mappedDatabases.add(mapping.transformOutboundDatabaseName(database));
+      }
+    }
+    return mappedDatabases;
+  }
+
+  public static void shutdownExecutorService(ExecutorService executorService) {
+    executorService.shutdown();
+    try {
+      if (!executorService.awaitTermination(200, TimeUnit.MILLISECONDS)) {
+        executorService.shutdownNow();
+      }
+    } catch (InterruptedException e) {
+      executorService.shutdownNow();
+    }
+  }
+
+  private static List<?> getResultFromFuture(Iterator<DatabaseMapping> iterator, Future<List<?>> future)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    DatabaseMapping mapping = iterator.next();
+    long timeout = mapping.getTimeout();
+    return future.get(timeout, TimeUnit.MILLISECONDS);
+  }
+
+  public static void getDatabasesFromFuture(List<Future<List<?>>> futures, Iterator<DatabaseMapping> iterator,
+                                            List<String> combined, String errorMessage, Logger log) {
+    for (Future<List<?>> future : futures) {
+      try {
+        List<String> result = (List<String>) getResultFromFuture(iterator, future);
+        combined.addAll(result);
+      } catch (InterruptedException e) {
```

Comment:
Is there a reason you're not logging the full stack trace for any of these? Do you really need separate messages? If not you could catch all three using `|` and have a more generic `log.warn("Unexpected error getting results", e);`

Reply:
No reason, just because it was like that (e.g.
```java
catch (TException e) {
   LOG.warn("Got exception fetching get_table_meta: {}", e.getMessage());
}
```
from PrefixBasedDatabaseMappingService, master version)

I initially had it for all errors, but Patrick added [this comment](https://github.com/HotelsDotCom/waggle-dance/pull/156#discussion_r261530103) and it made sense to me :)